internal/pkg/infrastructure/repositories/models/models.go

```diff
@@ -8,6 +8,7 @@ import (
 
 type WaterConsumption struct {
 	gorm.Model
+	WCOID       string `gorm:"index:idx_wco_id,unique"`
 	Device      string
```

Comment:
While we are at it we should add an index for the device/timestamp combo as well. No point in accepting multiple reading for the same device on the same time.